auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipalImpl.java

```diff
@@ -206,6 +206,42 @@ public String getName() {
         return roles;
     }
 
+    /**
+     * Get's the user realm roles
+     *
+     * @return user's realm roles
+     */
+    @Override
+    public Set<UserRole> getRealmRoles() {
+        Set<UserRole> realmRoles = new HashSet<>();
+        Iterator<UserRole> itr = roles.iterator();
+        while (itr.hasNext()) {
+            UserRole role = itr.next();
+            if (role.getType().equals(RoleType.REALM)) {
+                realmRoles.add(role);
+            }
+        }
+        return realmRoles;
+    }
+
+    /**
+     * Get's the user resource roles
+     *
+     * @return user's resource roles
+     */
+    @Override
+    public Set<UserRole> getResourceRoles() {
```

Comment:
those 2 methods are the same, except the RoleType. We can make that a parameter and a single method can used by both methods here.

Reply:
Added this single method there